src/mqt/bench/benchmark_generator.py

```diff
@@ -3,429 +3,382 @@
-                if not res:
-                    break
+        if not success_generated_circuits_t_indep:
+            return False
 
-        elif benchmark["name"] == "groundstate":
-            for choice in benchmark["instances"]:
-                res_qc_creation = qc_creation_watcher(create_groundstate_qc, [choice])
-                if not res_qc_creation:
-                    break
-                res = generate_circuits_on_all_levels(*res_qc_creation)
-                if not res:
-                    break
+        self.generate_target_dep_level_circuit(qc, num_qubits, file_precheck)
+        return True
 
-        elif benchmark["name"] == "pricingcall":
-            for nodes in range(benchmark["min_uncertainty"], benchmark["max_uncertainty"]):
-                res_qc_creation = qc_creation_watcher(create_pricingcall_qc, [nodes])
-                if not res_qc_creation:
-                    break
-                res = generate_circuits_on_all_levels(*res_qc_creation)
-                if not res:
-                    break
+    def generate_target_indep_level_circuit(self, qc: QuantumCircuit, num_qubits: int, file_precheck):
+        num_generated_circuits = 0
+        res_indep_qiskit = self.benchmark_generation_watcher(
+            qiskit_helper.get_indep_level,
+            self.timeout,
+            [qc, num_qubits, file_precheck, False, self.qasm_output_path],
+        )
+        if res_indep_qiskit:
+            num_generated_circuits += 1
 
-        elif benchmark["name"] == "pricingput":
-            for nodes in range(benchmark["min_uncertainty"], benchmark["max_uncertainty"]):
-                res_qc_creation = qc_creation_watcher(create_pricingput_qc, [nodes])
-                if not res_qc_creation:
-                    break
-                res = generate_circuits_on_all_levels(*res_qc_creation)
-                if not res:
-                    break
-        else:
-            for n in range(benchmark["min_qubits"], benchmark["max_qubits"], benchmark["stepsize"]):
-                res_qc_creation = qc_creation_watcher(create_scalable_qc, [benchmark, n])
-                if not res_qc_creation:
-                    break
-                res = generate_circuits_on_all_levels(*res_qc_creation)
+        res_indep_tket = self.benchmark_generation_watcher(
+            tket_helper.get_indep_level,
+            self.timeout,
+            [qc, num_qubits, file_precheck, False, self.qasm_output_path],
+        )
+        if res_indep_tket:
+            num_generated_circuits += 1
+
+        return num_generated_circuits != 0
+
+    def generate_target_dep_level_circuit(self, qc: QuantumCircuit, num_qubits: int, file_precheck):
+        compilation_paths = [
+            ("ibm", [("ibm_washington", 127), ("ibm_montreal", 27)]),
+            ("rigetti", [("rigetti_aspen_m2", 80)]),
+            ("ionq", [("ionq11", 11)]),
+            ("oqc", [("oqc_lucy", 8)]),
```

Comment:
Nothing for this PR, but this is something I'd definitely like to see improved in the future.
The hard-coding makes it fairly likely to break in the future and harder to extend.
If at some point the changes from that Predictor PR I started find their way in here, it should be possible to do this in a more programmatic function.

Reply:
I fully agree.